### ISSUE TITLE
[BUG] The M115 command does not work

### ISSUE THREAD

**Issue by radek8** [MarlinFirmware/Marlin#17922]
The M115 command does not work in the current version of Marlin bugfix 2.0

if allowed
#define M115_GEOMETRY_REPORT

related to the repair #16741

**Comment by thisiskeithb** [comment]
Please follow the [Issue Template](https://raw.githubusercontent.com/MarlinFirmware/Marlin/bugfix-2.0.x/.github/issue_template.md) when filing a bug report so we don't have to hunt for information.

```md
### Description

<!-- Description of the bug or requested feature -->

### Steps to Reproduce

<!-- If this is a Bug Report, please describe the steps needed to reproduce the issue -->

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:** [What you expect to happen]

**Actual behavior:** [What actually happens]

#### Additional Information

* Include a ZIP file containing your `Configuration.h` and `Configuration_adv.h` files.
* Provide pictures or links to videos that clearly demonstrate the issue.
* See [How Can I Contribute](#how-can-i-contribute) for additional guidelines.
```

**Comment by radek8** [comment]
A device that communicates via the serial port and calls the M115 command, for example the TFT display, does not work when the function is enabled
#define M115_GEOMETRY_REPORT


**Expected behavior:**

![M115 dis](https://user-images.githubusercontent.com/46979052/81479612-853bfb00-9224-11ea-8998-2726ed3e5430.JPG)

**Actual behavior:**

![M115 eror](https://user-images.githubusercontent.com/46979052/81479617-9127bd00-9224-11ea-9306-d70993cffb8f.JPG)

**Comment by sjasonsmith** [comment]
Post your configs.

**Comment by thisiskeithb** [comment]
*Please* follow the Issue Template.

**Comment by radek8** [comment]
> Post your configs.


no need to send configuration
by commenting #define M115_GEOMETRY_REPORT
output M115 is OK
@guruathwal should check if the settings work as intended.

**Comment by sjasonsmith** [comment]
Every issue should have configs posted. Not posting configs requires someone to waste their time creating their own configs which may or may not exhibit the same issue.

If you won’t spend the time to post your configs, why should someone else spend their time trying to help?

**Comment by radek8** [comment]
Configuration

[config.zip](https://github.com/MarlinFirmware/Marlin/files/4604566/config.zip)

**Comment by sjasonsmith** [comment]
OK, I've reproduced it. That is some pretty hilarious output. Far more entertaining than I expected.

**Comment by sjasonsmith** [comment]
There was an infinite loop in some recently added print macros. I've posted a PR to fix it.
@radek8 if you need to use this feature you can look at the PR linked above and easily implement the change yourself.

**Comment by sjasonsmith** [comment]
@radek8 the fix for this has been merged into bugfix-2.0.x. Please try it out and close the issue once you verify it is fixed.
